src/leap/mail/imap/mailbox.py

```diff
@@ -326,41 +322,7 @@ def addMessage(self, message, flags, date=None):
         :return: a deferred that will be triggered with the UID of the added
                  message.
         """
-        # TODO should raise ReadOnlyMailbox if not rw.
-        # TODO have a look at the cases for internal date in the rfc
-        if isinstance(message, (cStringIO.OutputType, StringIO.StringIO)):
-            message = message.getvalue()
-
-        # XXX we could treat the message as an IMessage from here
-        leap_assert_type(message, basestring)
-
-        if flags is None:
-            flags = tuple()
-        else:
-            flags = tuple(str(flag) for flag in flags)
-
-        if date is None:
-            date = formatdate(time.time())
-
-        # A better place for this would be  the COPY/APPEND dispatcher
-        # if PROFILE_CMD:
-        # do_profile_cmd(d, "APPEND")
-
-        # just_mdoc=True: feels HACKY, but improves a *lot* the responsiveness
-        # of the APPENDS: we just need to be notified when the mdoc
-        # is saved, and let's hope that the other parts are doing just fine.
-        # This will not catch any errors when the inserts of the other parts
-        # fail, but on the other hand allows us to return very quickly, which
-        # seems a good compromise given that we have to serialize the appends.
-        # A better solution will probably involve implementing MULTIAPPEND
-        # or patching imap server to support pipelining.
-
-        d = self.collection.add_msg(message, flags=flags, date=date,
-                                    notify_just_mdoc=True)
-
-        # XXX signal to UI? --- should do it only if INBOX...
```

Comment:
can you leave this XXX line there? it's something that still needs to be addressed
also, beware that you're removing the notify_just_mdoc here.